src/monitors/WhammMonitor.v3

```diff
@@ -66,46 +70,226 @@ class WhammMonitor(whamm: Module) extends Monitor {
 			}
 		}
 	}
+  // monitors called on parsed modules
 	def onParse(module: Module, err: ErrorGen) {
+	  this.err = err;
+	    // What extensions?
 		var wi = getWhammInstance(module).0;
 		if (wi == null) return;
 		if (onParseF >= 0) {
 			def moduleRef = ModuleRef.new(module);
 			Execute.call(wi.functions[onParseF], [Value.Ref(moduleRef)]); // TODO: report traps
 		}
 		if (opcodePatterns == null) return;
-		var it = BytecodeIterator.new();
 		for (i < module.functions.length) {
+		    // iterate over the functions in the module
 			var func = module.functions[i];
 			if (func.imp != null) continue; // imported
-			for (bi = it.reset(func); bi.more(); bi.next()) {
-				var bp = opcodePatterns[bi.current().tag];
-				for (l = bp; l != null; l = l.tail) {
-					if (Trace.whamm) Trace.OUT.put3("  match func %q+%d: %s", func.render(module.names, _), bi.pc, bi.current().mnemonic).ln();
-					var wf = WasmFunction.!(wi.functions[bp.head.2.func_index]);
-					var args = bindParams(it, bp.head.1);
-					Instrumentation.insertLocalProbe(module, func.func_index, bi.pc, WhammProbe.new(wf, args));
-				}
-			}
+
+      // parsed module is already validated, so we don't expect a new error here
+	    var cv = CodeValidator.new(Extension.WIZENG, Limits.new(), module, ErrorGen.new(""));
```

Comment:
I think we should pass in the outer error gen; even if we expect nothing to go wrong, it's probably better to not swallow any errors here.